packages/components/src/Datalist/Datalist.component.js

```diff
@@ -64,7 +64,18 @@ class Datalist extends Component {
 		const { value, previousValue } = this.state;
 
 		if (value !== previousValue) {
-			this.updateValue(event, value, true);
+			let newValue = value;
+			if (this.props.restricted && !this.props.multiSection && this.state.suggestions && value) {
+				const hasMatchingSuggestion = this.state.suggestions.find(
```

Comment:
You should base your search on titleMap props, not suggestions. 
Here it means that it doesn't take the suggestion if you close suggestions.